File: ABC310/b.cpp
// https://atcoder.jp/contests/abc310/tasks/abc310_b

#include <bits/stdc++.h>
using namespace std

int main(void) {
	int N, M;
	cin >> N >> M;

	vector<vector<int>> data(N, vector<int>(M+2, 0));
	for (int i = 0; i < N; i++) {
		cin >> data[i][0] >> data[i][1];

		int f_max = data[i][1];
		for (int j = 0; j < f_max; j++) {
			cin >> data[j+2];
		}
	}

	for (int i = 0; i < N; i++) {
		for (int j = i+1; j < N; j++) {
			if (data[i][0] <= data[j][0] && data[i][1] >= data[j][1]) { // dataの比較対象のどっちの値段が高いかがあらかじめわかるわけではない、、、
				if (check_)
			}
			else if ()
		}
	}
}

File: ABC310/b.py
# https://atcoder.jp/contests/abc310/tasks/abc310_b

"""
7/17 (月)
"""

# N, M = map(int, input().split())
# P, C, F = [], [], []
# for _ in range(N):
# 	l = list(map(int, input().split()))
# 	P.append(l[0])
# 	C.append(l[1])
# 	F.append(set(l[2:]))
# ans = False
# for i in range(N):
#     for j in range(N):
#         ans |= P[i] >= P[j] and F[j].issuperset(F[i]) \
#             and (P[i] > P[j] or len(F[j]) > len(F[i]))
# print('Yes' if ans else 'No')

"""
7/31(月)
"""

N, M = map(int, input().split())
P, C, F = [], [], []
for i in range(N):
    l = list(map(int, input().split()))
    P.append(l[0])
    C.append(l[1])
    F.append(set(l[2:]))
ans = False
for i in range(N):
    for j in range(N):
        ans |= P[i] >= P[j] and F[j].issuperset(F[i]) and (P[i] > P[j] or C[j] > C[i])
print("Yes" if ans else "No")


"""
7/18(火)
"""

# N, M = map(int, input().split()) # map(callable, iterable, *iterables) 返り値: function を iterable の全ての要素に適用して結果を生成 (yield) するイテレータ
#
# P, C, F = [], [], []
# for i in range(N):
#     l = list(map(int, input().split()))
#     P.append(l[0]);
#     C.append(l[1]);
#     F.append(set(l[2:]));
#
# ans = False:
# for i in range(N):
#     for j in range(N):
#         ans |= P[i] >= P[j] and F[j].issuperset(F[i]) \
#             and (P[i] > P[j] or C[j] > C[i])
#
# print('Yes' if ans else 'No')
